R/plotAbundanceDensity.R

```diff
@@ -0,0 +1,163 @@
+        # Gets data that will be plotted. Gets a list
+        density_data_list <- .incorporate_density_data(object = object,
+                                                       abund_values = abund_values,
+                                                       n = n,
+                                                       colour_by = colour_by)
+        # Extracts the density data and aesthetic from the list
+        density_data <- density_data_list$density_data
+        aesthetic <- density_data_list$aesthetic
+        
+        # Gets the plot from plotter
+        plot_out <- .density_plotter(density_data = density_data, 
+                                     aesthetic = aesthetic,
+                                     abund_values = abund_values, 
+                                     colour_by = colour_by)
+        return(plot_out)
+    }
+)
+
+################################ HELP FUNCTIONS ################################
+
+.incorporate_density_data <- function(object, abund_values, n, colour_by){
+    # Gets the assay
+    abundance_table <- assay(object, abund_values, withDimnames=FALSE)
+    # Gets the most abundant taxa
+    top_taxa <- getTopTaxa(object, n)
+    # Subsets abundance table  by taking taxa of highest abundance
+    abundance_table_subset <- abundance_table[top_taxa, , drop=FALSE]
+    
+    # Converts top taxa names to factor
+    top_taxa <- factor(top_taxa, rev(top_taxa))
+    
+    # Creates a data frame that includes sample names, taxa names as factors,
+    # and abundance values
+    density_data <- data.frame(
+        # Sample data = sample 1 is repeated n(taxa) times, sample 2 repeated n(taxaa) times...
+        sample = rep(seq_len(ncol(abundance_table_subset)), each = nrow(abundance_table_subset)), 
+        # taxa data = taxa 1, taxa 2,.. taxa n, taxa 1, taxa 2,... taxa n, as many times as there are samples
+        taxa = rep(top_taxa, ncol(abundance_table_subset)),
+        # Values grom abundance table
+        value = as.numeric(abundance_table_subset)
+    )
+    
+    # Gets coloring information if 'colour_by' is not NULL, otherwise specifies just aesthetic
+    if (!is.null(colour_by)) {
+        # Gets information from colData
+        colour_out <- retrieveCellInfo(object, colour_by)
+        # Mirrors back the variable name, if a partial match was used
+        colour_by <- colour_out$name
+        # Adds information to the data frame that includes all density data
+        density_data$colour_by <- colour_out$val[density_data$sample]
+        # Specifies aesthetic
+        aesthetic <- aes_string(y = "taxa", x = "value", colour = "colour_by")
+    } else {
+        # Specifies aesthetic, when colour_by is not used
+        aesthetic <- aes_string(y = "taxa", x = "value")
+    }
+    
+    # Creates a list that includes the data and aesthetic
+    density_data_list <- list(density_data = density_data, 
+                              aesthetic = aesthetic)
+    return(density_data_list)
+}
```

Comment:
As mentioned above, this functions shouldn't deal with the aesthetics.
 In addition I think it can be shortened a bit
```r
# Gets the assay
    mat <- assay(object, abund_values)
    # Gets the most abundant taxa
    top_taxa <- getTopTaxa(object, n)
    # Subsets abundance table  by taking taxa of highest abundance
    mat <- mat[top_taxa, , drop=FALSE]
    # melt the data
    plot_data <- t(mat) %>%
        as.data.frame() %>%
        rownames_to_column("Sample") 
    # Gets coloring information if 'colour_by' is not NULL, otherwise specifies just aesthetic
    if (!is.null(colour_by)) {
        # Gets information from colData
        colour_out <- retrieveCellInfo(object, colour_by)
        # Mirrors back the variable name, if a partial match was used
        colour_by <- colour_out$name
        # Adds information to the data frame that includes all density data
        plot_data$colour_by <- colour_out$value
    }
    cols <- c("Sample","colour_by")[c("Sample","colour_by") %in% colnames(plot_data)]
    plot_data <- plot_data %>%
        pivot_longer(cols = !cols, names_to = "Y", values_to = "X")
    return(list(plot_data = plot_data, 
                colour_by = colour_by))
```